src/shared/sleep-config.c

```diff
@@ -310,6 +349,29 @@ int find_hibernate_location(HibernateLocation **ret_hibernate_location) {
         if (r < 0)
                 return r;
 
+        /* Users using swapfiles with Btrfs may want to override this check - especially if they
+         * are using multiple swapfiles or other complex swap configurations.
+         */
+        if (getenv_bool("SYSTEMD_BYPASS_HIBERNATION_MEMORY_CHECK") > 0) {
+                if (sys_resume == 0)
+                        return log_warning_errno(SYNTHETIC_ERRNO(ENOSYS),
+                                                 "SYSTEMD_BYPASS_HIBERNATION_MEMORY_CHECK is set but /sys/power/resume is not configured; hibernation will fail");
+
+                hibernate_location = new(HibernateLocation, 1);
+                if (!hibernate_location)
+                        return log_oom();
+
+                *hibernate_location = (HibernateLocation) {
+                        .resume = sys_resume,
+                        .resume_offset = sys_offset,
+                        .swap = NULL,
```

Comment:
doesn't matter, but this line is redundant. Fields that aren't specified are initialized to 0 anyway...